src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt

```diff
@@ -91,6 +91,17 @@ open class NexusClient(private val baseUrl: URI, username: String?, password: St
             ?.id
     }
 
+    fun findStagingRepository(stagingProfileId: String, descriptionRegex: Regex): StagingRepositoryDescriptor {
+        val response = api.getStagingRepositories(stagingProfileId).execute()
+        if (!response.isSuccessful) {
+            throw failure("find staging repository, stagingProfileId: $stagingProfileId, descriptionRegex: $descriptionRegex", response)
+        }
+        val totalNumberOfFoundRepos = response.body()?.data?.count()
+        return response.body()?.data?.firstOrNull { it.description?.contains(descriptionRegex) == true }
```

Comment:
> We should probably define what happens if multiple repositories match.
> 
> We might add an enum like: FAIL (default), USE_LATEST

FAIL is much safer and I agree should be a default behavior. What are the cases for `USE_LATEST`? Some previous execution errors with leftover staging repositories? I wonder, if a developer should be forced to log in to the Nexus UI to clean it up, or you believe that USE_LATEST should be "safe" (if explicitly chosen)?

Btw, I still assume that without explicitly calling the `find*StagingRepository` task, the staging repository id should be called manually from the command line (if init and close/release are not performed in the same execution), right?

(move to the inline command to make it easier to follow)

Reply:
I've hard-coded "fail if multiple repositories found"